code/lib/cli/src/proxy.ts

```diff
@@ -0,0 +1,21 @@
+import { spawn } from 'child_process';
+import { versions } from '@storybook/core/common';
+
+const args = process.argv.slice(2);
+
+if (['dev', 'build'].includes(args[0])) {
+  require('@storybook/core/cli/bin');
+} else {
+  const proxiedArgs =
+    args[0] === 'init'
+      ? [`create-storybook@${versions.storybook}`, ...args.slice(1)]
+      : [`@storybook/cli@${versions.storybook}`, ...args];
+  const command = ['npx', '--yes', ...proxiedArgs];
```

Comment:
Addressing @IanVS' concern, we should use pnpm/yarn/npm equivalent commands here

Reply:
@JReinhold I wonder what you think about this

Reply:
I'm not sure it makes an actual difference in this case? Especially given that the underlying commands still auto-detects the package manager. But I could be wrong.
Perhaps it stops it from reusing an already globally cached version of the CLI if it was cached by pnpm/yarn, but that's the only thing I can see.

Reply:
Also, we won't recommend the forwarding solution for `init`. We will recommend from 8.3 to use:

* `pnpm create storybook@latest`
* `yarn create storybook@latest`
* `npm create storybook@latest`

For the ones that we recommend the forwarding are (among others)
* `npx storybook migrate ...`
* `npx storybook automigrate`
* `npx storybook doctor`
etc.

And for those currently, we also don't recommend `pnpm dlx` as this will not get the local installed package.

Reply:
I think the most unknown case is upgrade.

I'm not sure if there would be slight differences between the old and new:
`pnpm dlx storybook@latest upgrade`

To circumvent that we could recommend:
`pnpm dlx @storybook/cli@latest upgrade`

Which doesn't forward.

Reply:
>And for those currently, we also don't recommend pnpm dlx as this will not get the local installed package.

To run the locally-installed version, I would use `pnpm storybook migrate...`, etc.  I basically never want to run `npx` in a pnpm project, I've had wonky results before in the past (though I can't remember the details).

Reply:
This is a good point. But at this moment `sb init` is making a script `storybook` that is actually calling `storybook dev`. We should probably change that though. I do agree with @IanVS that it would be ideal if every user had a script `storybook` that just pointed to the local `storybook` executable.

cc @shilman @ndelangen

Reply:
This SGTM, we can resolve the conversation!

Reply:
@kasperpeulen you don't even need a script to point to the local, so long as you haven't overloaded it I guess.  Yarn and pnpm are similar in that you can just `yarn foo` or `pnpm foo` and it will run the `foo` dependency that is locally installed.

Reply:
I see so we just have to make sure that our sb init script changes those names:
<img width="802" alt="image" src="https://github.com/user-attachments/assets/68b09bbe-28a2-4c66-be75-5a7a9d88f87a">

to something like `storybook:dev` and `storybook:build`

Reply:
Yeah, that would work, though I'm not sure it's worth it, since folks are so used to these commands it's muscle-memory by now I'm sure.  The other option is to change the name of the storybook executable to `@storybook/cli` or something, which I think you suggested above.

Reply:
@kasperpeulen changing what is supposed to be set in the `package.json` `scripts` seems not really do-able in a minor release?

It's such a impactful change, it would really take a lot of effort; don't you think?
- All users would have to learn to run a different command
- Chromatic must be changed
- We'd need some sort of auto-migration

Checking if I understand the problem correctly:
Some package managers way of running a dependency's bin is the same as running a scripts.
This hinders users from running some command.

Is upgrade a problem?
User's running upgrade will never upgrade to their current version, right?
So won't they add a version specifying to the command? 
Does this not force the package manager to not use a local version, and not use the script?
If a user runs this:
> `pnpm dlx storybook@latest upgrade`
Then it should just work, right?
Why would this not work?

Is `init` a problem?
It can't be, because then the local script is missing, and with the new method, it's even clearer,; right?
> `pnpm create storybook@latest`

Is `dev` a problem?
Well if it uses the script, that's exactly what we want.

Is `build` a problem?
Nope that's always been it's own separate script.

Is `doctor` a problem?
?

Is `automigrate` a problem?
?

Reply:
To be clear, I also think that changing the script name is not a good idea.

>If a user runs this:
>
>>pnpm dlx storybook@latest upgrade
>
>Then it should just work, right?
>Why would this not work?

`pnpm dlx` downloads and runs the latest version if no version is specified (unlike `npx`), so the `@latest` is unnecessary there.

>Is doctor a problem?
>?
>
>Is automigrate a problem?
>?

Yes I think it is these kinds of commands we are discussing.  But I just tested, and while `pnpm storybook doctor` does not work (it tries to run `storybook dev doctor`, what _does_ work is `pnpm exec storybook doctor` (at least when using pnpm 8.14.2).  So, I think maybe no changes are really required on storybook's side, other than perhaps updating some documentation to make it clear how to run everything, and if `npx` is being used to run them as part of other scripts, maybe we could detect the package manager and run the appropriate other command like `pnpm exec` instead.

Reply:
> if `npx` is being used to run them as part of other scripts, maybe we could detect the package manager and run the appropriate other command like `pnpm exec` instead.

What would be the reason to do that?

We could do that detection, but if `npx` just always works in the right way, that will be considerable less complexity, and things that could go wrong.

Reply:
I'm not sure, to be honest.  As a `pnpm` user, I do my best not to use `npx`, because it feels like mixing package managers.  Maybe it's fine, it just makes me nervous.